Toggl.Core.Tests/Interactors/Sync/PushSyncInteractorTests.cs

```diff
@@ -33,6 +39,177 @@ public sealed class TheConstructor : BaseInteractorTests
 
         public sealed class TheExecuteMethod : BaseInteractorTests
         {
+            private readonly IInteractorFactory mockInteractorFactory = Substitute.For<IInteractorFactory>();
+            private readonly PushSyncInteractor interactor;
+            private readonly Request req;
+
+            public TheExecuteMethod()
+            {
+                interactor = new PushSyncInteractor(
+                    Api, Database.PushRequestIdentifier, mockInteractorFactory, QueryFactory);
+
+                req = new Request("user agent name");
+                req.CreateTags(new[] { new MockTag() }); // just to make sure that it's not empty
+                var prepareInteractor = Substitute.For<IInteractor<Task<Request>>>();
+                var task = Task.FromResult(req);
+                prepareInteractor.Execute().Returns(task);
+                mockInteractorFactory.PreparePushRequest().Returns(prepareInteractor);
+            }
+
+            [Fact, LogIfTooSlow]
+            public async Task DoesNotMakeAHttpRequestIfThereIsNothingToPush()
+            {
+                var emptyReq = new Request("user agent name");
+                var task = Task.FromResult(emptyReq);
+                var prepareInteractor = Substitute.For<IInteractor<Task<Request>>>();
+                prepareInteractor.Execute().Returns(task);
+                mockInteractorFactory.PreparePushRequest().Returns(prepareInteractor);
+
+                await interactor.Execute();
+
+                await Api.SyncApi.DidNotReceive().Push(Arg.Any<Guid>(), Arg.Any<Request>());
+            }
+
+            [Fact, LogIfTooSlow]
+            public async Task SendsTheDirtyDataToTheSyncServer()
+            {
+                await interactor.Execute();
+
+                await Api.SyncApi.Received().Push(Arg.Any<Guid>(), Arg.Is(req));
+            }
+
+            [Fact, LogIfTooSlow]
+            public async Task StoresResultsInTheDatabase()
+            {
+                var res = Substitute.For<IResponse>();
+                var throwingTask = Task.FromResult(res);
+                Api.SyncApi.Push(Arg.Any<Guid>(), Arg.Any<Request>()).Returns(throwingTask);
+
+                await interactor.Execute();
+
+                QueryFactory.Received().ProcessPushResult(Arg.Is(res));
```

Comment:
I think this works normally like this as well:

```suggestion
                QueryFactory.Received().ProcessPushResult(res);
```

Reply:
In this case I prefer using `Arg.Is`, let's keep it 😄